test/script.clj

```diff
@@ -35,10 +36,59 @@
 (def ev2 @events)
 
 (deftest events-test
-  (is (pos? (count ev1)))
-  (is (contains? (set (map :path ev1)) "test/script.clj"))
+  (is (= 1 (count ev1)))
+  (is (= (:path (first ev1)) "test/script.clj"))
   (testing "No new events after unwatch"
     (is (= (count ev1) (count ev2)))))
 
+(deftest dedup-test
+  (reset! events [])
+  (let [watcher (fw/watch "test" #(swap! events conj %) {:delay-ms 50 :recursive true})]
+    (sh "touch" *file*)
+    (Thread/sleep 5)
+    (sh "touch" *file*)
+    (Thread/sleep 5)
+    (sh "touch" *file*)
+    ;;wait for timer to end
+    (Thread/sleep 60)
+    (prn :events-dedup @events)
+    (testing "tests that the events that happened inside the interval were deduped."
+      (is (= 1 (count @events))))
+    (fw/unwatch watcher)))
+
+(deftest dedup-outside-interval-test
```

Comment:
this test covers multiple notification for the same file, outside the dedup interval.